Packs/ExtraHop/Integrations/ExtrahopRevealXEventCollector/ExtrahopRevealXEventCollector.py

```diff
@@ -0,0 +1,471 @@
+        mod_time = detection.get("mod_time")
+        start_time = detection.get("start_time")
+        update_time = detection.get("update_time")
+
+        if mod_time:
+            detection["_TIME"] = mod_time
+            if start_time:
+                if mod_time == start_time:
+                    detection["_ENTRY_STATUS"] = "new"
+                elif mod_time > start_time or (update_time and update_time > start_time):
+                    detection["_ENTRY_STATUS"] = "updated"
+
+
+def get_detections_list(client: Client,
+                            advanced_filter=None) -> List[dict[str, Any]]:
+    """Retrieve the detections from ExtraHop-Reveal(X).
+
+    Args:
+        client: ExtraHop client to be used.
+        advanced_filter: The advanced filter provided by user to fetch detections.
+
+    Returns:
+        CommandResults object.
+    """
+    body = advanced_filter
+    trim_spaces_from_args(body)
+    detections = list(client.detections_list(body))
+    # Add Time Params
+    update_time_values_detections(detections)
+    return detections
+
+
+def fetch_extrahop_detections(
+    client: Client,
+    advanced_filter: dict,
+    last_run: dict,
+    max_events: int,
+) -> tuple[List, dict]:
+    """
+    Fetch detections from ExtraHop according to the given filter.
+
+    Args:
+        max_events: Maximum number of events to fetch.
+        client: ExtraHop client to be used.
+        advanced_filter: The advanced_filter given by the user to filter out the required detections.
+        last_run: Last run returned by function demisto.getLastRun.
+
+    Returns:
+        List of incidents to be pushed into XSIAM.
+    """
+    try:
+        already_fetched: List = last_run.get("already_fetched", [])
+        detection_start_time = advanced_filter["mod_time"]
+        events = []
+
+        while True:
+            detections = get_detections_list(client, advanced_filter=advanced_filter)
+
+            if not detections:
+                # Didn't get any detections or got all duplicates
+                break
+
+            # Check if all detections are already fetched
+            new_detections = [detection for detection in detections if detection.get("id") not in already_fetched]
+
+            if not new_detections:
+                # If no new detections, break the loop to prevent an infinite loop
+                demisto.info("No new detections to fetch, exiting the loop.")
+                break
+
+            for detection in new_detections:
+                detection_id = detection.get("id")
+                already_fetched.append(detection_id)
+                events.append(detection)
+
+                if len(events) == max_events:
+                    break
+
+            # hit max_events
+            if events and len(events) == max_events:
+                # Compare mod_time of last event with the first in this batch
+                if new_detections and events[-1]["mod_time"] == detections[0]["mod_time"]:
```

Comment:
All this logic will be more clear if you used:
                last_event = events[-1]
                first_event = detections[0]